File: examples/java/testing/examples/sequential-workflow.java
import static org.junit.jupiter.api.Assertions.*;

import java.util.Map;
import org.junit.jupiter.api.Test;
import software.amazon.awssdk.services.lambda.model.OperationType;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.model.ExecutionStatus;
import software.amazon.lambda.durable.testing.LocalDurableTestRunner;

class SequentialWorkflowTest {

    record Order(String orderId) {}

    @Test
    void executesAllStepsInOrder() {
        var runner = LocalDurableTestRunner.create(
            Order.class,
            (input, context) -> {
                var validated = context.step("validate", Map.class,
                    ctx -> Map.of("orderId", input.orderId(), "status", "validated"));
                var paid = context.step("payment", Map.class,
                    ctx -> Map.of("orderId", input.orderId(), "payment", "completed"));
                return context.step("fulfillment", Map.class,
                    ctx -> Map.of("orderId", input.orderId(), "fulfillment", "shipped"));
            }
        );

        var result = runner.runUntilComplete(new Order("order-123"));

        assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

        var stepOps = result.getOperations().stream()
            .filter(op -> op.getType() == OperationType.STEP)
            .toList();
        assertEquals(3, stepOps.size());
        assertEquals("validate", stepOps.get(0).getName());
        assertEquals("payment", stepOps.get(1).getName());
        assertEquals("fulfillment", stepOps.get(2).getName());
    }
}

File: examples/csharp/testing/examples/sequential-workflow.cs
using System.Linq;
using Amazon.Lambda.DurableExecution;
using Amazon.Lambda.DurableExecution.Testing;
using Xunit;

public class SequentialWorkflowTest
{
    public record Order(string OrderId, string? Status = null, string? Payment = null, string? Fulfillment = null);

    private static async Task<Order> Workflow(Order input, IDurableContext ctx)
    {
        Order validated = await ctx.StepAsync(
            async (_, _) => input with { Status = "validated" },
            name: "validate");
        Order paid = await ctx.StepAsync(
            async (_, _) => validated with { Payment = "completed" },
            name: "payment");
        return await ctx.StepAsync(
            async (_, _) => paid with { Fulfillment = "shipped" },
            name: "fulfillment");
    }

    [Fact]
    public async Task ExecutesAllStepsInOrder()
    {
        await using var runner = new DurableTestRunner<Order, Order>(Workflow);

        TestResult<Order> result = await runner.RunAsync(new Order("order-123"));

        Assert.True(result.IsSucceeded);

        var stepOps = result.Steps.Where(s => s.Kind == OperationKind.Step).ToList();
        Assert.Equal(3, stepOps.Count);
        Assert.Equal(
            new[] { "validate", "payment", "fulfillment" },
            stepOps.Select(s => s.Name).ToArray());
    }
}
